### ISSUE TITLE
The training with Model Builder is utterly slow

### ISSUE THREAD

**Issue by forReason** [dotnet/machinelearning-modelbuilder#993]
### System information

- Win 10, 1909
- .net core 3.1.402
- Intel core I7 3770k
- Nvidia RTX 2080

### Issue

- **What did you do?**
1. I created an image dataset similar to mnist, roughly 10000 images, each image 64x36 px, 2 classes
[dataset (~25mb download)](http://julianbechtold.de/media/SuperDownSample.zip)
2. Opened the model builder (first time to try ml.net), found no method to create your own model, so i choose image classification.
3. Started training

- **What happened?**
After > 1h of training and seeing in the log that each iteration of one image(!!!) takes roughly 0.5 seconds and it was still at the step of "Bottleneck computation", I stopped training
![image](https://user-images.githubusercontent.com/12736950/92920802-2f29cf00-f433-11ea-939f-1ed82be1412e.png)
Additionally I saw that a way oversized resnet with transfer learning was thrown at the task whereas I manually would have chosen a simple network of 1-2 Convolutional layers (if even) and a couple of normal connected layers.
Also, the GPU does not seem to be utilized, eventhough cuda drivers and cupti tools are installed
![image](https://user-images.githubusercontent.com/12736950/92922565-5766fd00-f436-11ea-9802-daa5219c6816.png)
![image](https://user-images.githubusercontent.com/12736950/92921495-6ba9fa80-f434-11ea-9985-e3cd0fd0f67a.png)
Later on, I noticed in the logs, that indeed, the Model reached an accuracy of 99% right after epoch 2 - sufficient for me (at least for a prototype). BUT automl tries to find the next model, goes on to the 10 Minutes lasting "Bottleneck computation" AGAIN and starts all over. 
![image](https://user-images.githubusercontent.com/12736950/92922476-31415d00-f436-11ea-8faa-d0dd7b2e08c8.png)
There is no way to say, "I want to reach accuracy X" or "train for X Minutes". When I press "Cancle Training", all tested models and training progress is discarded.
![image](https://user-images.githubusercontent.com/12736950/92922833-b9bffd80-f436-11ea-8bc0-5ca70abcecaa.png)

- **What did you expect?**
I expected training to take max 10 Minutes.
If I approached mnist with tensorflow, Id be going with a simple, ready model in roughly 10 Minutes. The main purpose of ml.net for me was the compatibility with c# but I was quite unimpressed to see the capabilities of the Model builder if I can only select 4 preset scenarios which I cannot even customize to my needs.

### Source code / logs
No soucecode so far, as I dont get beyond the training.

**Comment by forReason** [comment]
Thank you so much, great conclusion.

**Comment by forReason** [comment]
Notably, I want to say that I followed the "Ml.Net Get Started" tutorial from Microsoft along:
https://dotnet.microsoft.com/learn/ml-dotnet/get-started-tutorial/intro

The tutorial shows multiple of my requested Features, which have been removed (?):
![image](https://user-images.githubusercontent.com/12736950/93051765-46500300-f665-11ea-841a-08c49b49ea24.png)

**Comment by JakeRadMSFT** [comment]
@forReason Thanks for reaching out! We do have GPU support but it requires a little extra setup:

https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/install-gpu-model-builder

I'll review the rest of the comments today. Thanks for using Model Builder and ML.NET!

**Comment by forReason** [comment]
actually I would love to work on some stuff but it appears as if the project is closed source 👎

**Comment by LittleLittleCloud** [comment]
@forReason 
The code for image training is not closed-source. Model builder uses AutoML.Net for image training, which is open source. Here is the code.
https://github.com/dotnet/machinelearning/blob/5929292f4c45a330b2b2541bd59ff2138424a6ee/src/Microsoft.ML.AutoML/TrainerExtensions/MultiTrainerExtensions.cs#L243

About the ability to set up maximum training time, These features are only available in tabular-data training in model builder, and we decide to disable these features for image training because it doesn't support save training status and retrain, which means if the training doesn't finish before time is used up, the user will need to start from the very beginning and that could be a huge frustrate. 

Another possible reason that your dataset cost ~3 hours is AutoML is actually running a 10 fold cross-validation for you, since your dataset only contains ~10000 pictures, which is lower than the stop-cv threshold.  That's also why you can see the bottleneck computation been raised for multiple times, (to be exact 10 times.) So the actual training time should be ~18 minutes, see the following code for more information. I would suggest mlnet simply disable the cross-validation for image training case, after all, it doesn't run any hyper-parameter tuning, thus I don't understand what's the meaning of run the same training with the same option for 10 times. thoughts? @harishsk @antoniovs1029 

https://github.com/dotnet/machinelearning/blob/5929292f4c45a330b2b2541bd59ff2138424a6ee/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L115

One more thing, AutoML.Net has early stop, but it only come into effect when score on validation dataset hit 100%. So to AutoML.Net, the training is simply not good enough :-P
